packages/teams-js/src/public/dialog.ts

```diff
@@ -88,10 +88,13 @@ export namespace dialog {
       registerHandler('messageForParent', messageFromChildHandler);
     }
     const dialogInfo: DialogInfo = getDialogInfoFromUrlDialogInfo(urlDialogInfo);
-    sendMessageToParent('tasks.startTask', [dialogInfo], (err: string, result: string | object) => {
-      submitHandler({ err, result });
-      removeHandler('messageForParent');
-    });
+    const submitHandlerHelper = submitHandler
+      ? (err: string, result: string | object) => {
+          submitHandler({ err, result });
+          removeHandler('messageForParent');
+        }
+      : () => removeHandler('messageForParent');
```

Comment:
Interestingly, In javascript/typescript we can call a function with more arguments than there are parameters. The extra arguments are simply ignored. 
https://www.typescriptlang.org/docs/handbook/2/functions.html

In case of no submitHandler, the callback will have zero parameters, but still be called with 2 arguments. And, it would work fine!